### PR TITLE
Functions: Completion and inspection for TimerTrigger schedule expression 

### LINKED ISSUE #296: Functions: Completion and inspection for TimerTrigger schedule expression
https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=csharp#ncrontab-expressions

### PR DESCRIPTION
**Pull request by maartenba**
Fixes #296 - thanks @khalidabuhakmeh for the suggestion!

image

### CODE REVIEW

**Review by sdubov — APPROVED** [review]
@maartenba, do you think it is a good time to add backend to our plugin? ;)

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/completion/csharp/AzureFunctionsTimerTriggerCompletionContributor.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                            parameters: CompletionParameters,
+                            context: ProcessingContext,
+                            resultSet: CompletionResultSet) {
+
+                        val element = parameters.position
+                        val timerTriggerAttributeElement = AzureFunctionsPsiHelper.tryResolveTimerTriggerAttributeFromScheduleExpressionStringLiteralSibling(element)
+                        if (timerTriggerAttributeElement != null) {
+                            for (cronSuggestion in cronSuggestions) {
+
+                                val cron = cronParser.parse(cronSuggestion)
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/completion/csharp/AzureFunctionsTimerTriggerCompletionContributor.kt]: Maybe better to handle exception on `parse()` here. Log and continue with other suggestions or throw after processing?

**Review by sdubov — CHANGES_REQUESTED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/completion/csharp/AzureFunctionsTimerTriggerCompletionContributor.kt (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/completion/csharp/AzureFunctionsTimerTriggerCompletionContributor.kt]: Did you get an exception on any of the suggestions? I'll add a check for it, but ideally it should not throw at all with the examples in the file :-)

**Comment by maartenba** [comment]
Backend would be a godsend, but no idea how to best structure the plugin then :-)

**Review by sdubov — APPROVED** [review]

**Comment by maartenba** [comment]
@sdubov Should this also be moved to backend? #307

Uses NCRONTAB expressions: https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=csharp#ncrontab-expressions (six-part expression type)
ncrontab library: https://www.nuget.org/packages/ncrontab/

One thing NCrontab does not seem to have is a way to "describe" the expression (e.g. when hovering)

**Comment by sdubov** [comment]
@maartenba, sure. Make sense - #310.

**Pull request merged**